featuretools/tests/primitive_tests/test_transform_features.py

```diff
@@ -1570,26 +1578,110 @@ def test_cfm_with_lag_and_non_nullable_column(pd_es):
         trans_primitives=[lag_primitive],
         cutoff_time=cutoff_times,
     )
-
-    # Non nullable
     assert fm["NUMERIC_LAG(datetime, value, periods=5)"].head(periods).isnull().all()
     assert fm["NUMERIC_LAG(datetime, value, periods=5)"].isnull().sum() == periods
-    # Nullable
+
     assert "NUMERIC_LAG(datetime, value_2, periods=5)" in fm.columns
+
+    assert "NUMERIC_LAG(datetime, products.rating, periods=5)" in fm.columns
     assert (
         fm["NUMERIC_LAG(datetime, products.rating, periods=5)"]
         .head(periods)
         .isnull()
         .all()
     )
 
-    assert "NUMERIC_LAG(datetime, products.rating, periods=5)" in fm.columns
+
+def test_cfm_with_lag_and_non_nullable_columns(pd_es):
+    # fill nans so we can use non nullable numeric logical type in the EntitySet
+    new_log = pd_es["log"].copy()
+    new_log["value"] = new_log["value"].fillna(0)
+    new_log["value_double"] = new_log["value"]
+    new_log["purchased_with_nulls"] = new_log["purchased"]
+    new_log["purchased_with_nulls"][0:4] = None
+    new_log.ww.init(
+        logical_types={
+            "value": "Integer",
+            "product_id": "Categorical",
+            "value_double": "Double",
+        },
+        index="id",
+        time_index="datetime",
+        name="new_log",
+    )
+    pd_es.add_dataframe(new_log)
+    rels = [
+        ("sessions", "id", "new_log", "session_id"),
+        ("products", "id", "new_log", "product_id"),
+    ]
+    pd_es = pd_es.add_relationships(rels)
+
+    assert isinstance(pd_es["new_log"].ww.logical_types["value"], Integer)
+
+    periods = 5
+    lag_primitive = Lag(periods=periods)
+    cutoff_times = pd_es["new_log"][["id", "datetime"]]
+    fm, _ = dfs(
+        target_dataframe_name="new_log",
+        entityset=pd_es,
+        agg_primitives=[],
+        trans_primitives=[lag_primitive],
+        cutoff_time=cutoff_times,
+    )
+    # Integer
+    assert fm["LAG(value, datetime, periods=5)"].head(periods).isnull().all()
+    assert fm["LAG(value, datetime, periods=5)"].isnull().sum() == periods
     assert (
```

Comment:
This isn't a big deal or a blocker in my opinion, but instead of checking the type string, you could also check that the logical type is an instance of the expected logical type class like is done on L1619.

Reply:
yup just was lazy about importing :P. 

Switched!